### ISSUE TITLE
Where could I find more info when Cicerone falied to apply command?

### ISSUE THREAD

**Issue by Kiloqky** [terrakok/Cicerone#160]
## Problem
Hello everyone, I have a crash in my app with stacktrace related to Cicerone. I tried to invistigate the reason but couldn't

Fatal Exception: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceStateat 
    androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1)at 
    androidx.fragment.app.FragmentManager.popBackStack(FragmentManager.java:1)at             
    com.github.terrakok.cicerone.androidx.AppNavigator.backToRoot(AppNavigator.java:1)at 
    com.github.terrakok.cicerone.androidx.AppNavigator.backTo(AppNavigator.java:15)at 
    com.github.terrakok.cicerone.androidx.AppNavigator.applyCommandsSync(AppNavigator.java:8)at 
    com.github.terrakok.cicerone.androidx.AppNavigator$applyCommands$1.run(AppNavigator.java:8)at 
    android.os.Handler.handleCallback(Handler.java:883)at android.os.Handler.dispatchMessage(Handler.java:100)at 
    android.os.Looper.loop(Looper.java:237)at android.app.ActivityThread.main(ActivityThread.java:8019)at 
    java.lang.reflect.Method.invoke(Method.java)at 
    com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)at 
    com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
    
    
## Actual behaviour
Stacktrace doesn't have information about `Screen` or `transition type`. I can't realize how to reproduce and fix it.
## Expected behaviour
I see information about `Screen` or `transition type` in the `Throwable message`. I'm able to reproduce and fix the throwable.

**Comment by terrakok** [comment]
@VitalyPeryatin I think we should check FragmentManager state before `applyCommandsSync` if it is possible.  
Looks like handler message was applied after state saving  
https://github.com/terrakok/Cicerone/commit/f2de564600e4a4422e5b0132b061550a81aa95a2

**Comment by VitalyPeryatin** [comment]
@terrakok I think check FragmentManager state on calling `onSaveInstanceState` is not possible. But we can solve the problem if we replace `handler.post { }` from `AppNavigator` to `CommandBuffer` and place it inside `executeCommands(Array<out Command>)`. Like this:
```
fun executeCommands(commands: Array<out Command>) {
        mainHandler.post {
            navigator?.applyCommands(commands) ?: pendingCommands.add(commands)
        }
}
```
It will work because `navigator` inside `CommandBuffer` is clearing in `onPause` (100% before calling `onSaveInstanceState`)

**Comment by terrakok** [comment]
Fixed in version 7.1
